src/zenml/integrations/gcp/orchestrators/vertex_orchestrator.py

```diff
@@ -363,77 +348,171 @@ def prepare_or_run_pipeline(
+                        dynamic_component.add_node_selector_constraint(
+                            label_name=key, value=value
+                        )
+
+                # add resource requirements
+                if step.config.resource_settings:
+                    cpu_limit = (
+                        step.config.resource_settings.cpu_count
+                        or self.config.cpu_limit
+                    )
 
-                for key, value in environment.items():
-                    container_op.set_env_variable(name=key, value=value)
-
-                # Set upstream tasks as a dependency of the current step
-                for upstream_step_name in step.spec.upstream_steps:
-                    upstream_container_op = step_name_to_container_op[
-                        upstream_step_name
-                    ]
-                    container_op.after(upstream_container_op)
+                if cpu_limit is not None:
+                    dynamic_component = dynamic_component.set_cpu_limit(
+                        str(cpu_limit)
+                    )
 
-                settings = cast(
-                    VertexOrchestratorSettings,
-                    self.get_settings(step),
+                memory_limit = (
+                    step.config.resource_settings.memory[:-1]
+                    if step.config.resource_settings.memory
+                    else self.config.memory_limit
                 )
-                if settings.pod_settings:
-                    apply_pod_settings(
-                        container_op=container_op,
-                        settings=settings.pod_settings,
+                if memory_limit is not None:
+                    dynamic_component = dynamic_component.set_memory_limit(
+                        memory_limit
                     )
 
-                self._configure_container_resources(
-                    container_op=container_op,
-                    resource_settings=step.config.resource_settings,
-                    node_selector_constraint=settings.node_selector_constraint,
+                gpu_limit = (
+                    step.config.resource_settings.gpu_count
+                    if step.config.resource_settings.gpu_count is not None
+                    else self.config.gpu_limit
                 )
-                container_op.set_caching_options(enable_caching=False)
+                if gpu_limit is not None and gpu_limit > 0:
+                    dynamic_component = dynamic_component.set_gpu_limit(
+                        gpu_limit
+                    )
+
+                if step_settings.node_selector_constraint:
+                    (constraint_label, value) = (
+                        step_settings.node_selector_constraint
+                    )
+                    if not (
+                        constraint_label
+                        == GKE_ACCELERATOR_NODE_SELECTOR_CONSTRAINT_LABEL
+                        and gpu_limit == 0
+                    ):
```

Comment:
I find this double conditional a bit hard to understand. Maybe simplify somehow, or add a comment stating exactly what is happening here?